### PR TITLE
Add limit and offset to query function and implement suggest function

### LINKED ISSUE #5: Implement search methods
- [x]  Define typespecs / behaviour for public functions
- [x] Search in the database `QUERY <collection> <bucket> "<terms>"`
- [x] Limit search items using `[LIMIT(<count>)]`
- [x] Offset search items using `[OFFSET(<count>)]`
- [x] Suggestion to autocomplete word `SUGGEST <collection> <bucket> "<word>" [LIMIT(<count>)]?`

### PR DESCRIPTION
**Pull request by edgarlatorre**
## ✍️ Description

Add limit and offset options to search and implement suggest function.

- Add limit and offset to search function
- Add suggest function to the search module which returns list of words to autocomplete

## ✅ Fixes

(optional)

- Closes #5

### CODE REVIEW

File: lib/modes/search.ex [hunk 1]
```diff
@@ -1,9 +1,63 @@
...
+  {:ok, ["object1", "object2"]}
+
+  iex> SonicClient.Modes.Search.query("my_collection", "bucket", "test", limit: 1)
+  {:ok, ["object1"]}
+
+  iex> SonicClient.Modes.Search.query("my_collection", "bucket", "test", limit: 1, offset: 1)
+  {:ok, ["object2"]}
+
+  """
+  def query(conn, collection, bucket, terms, opts \\ [limit: 10, offset: 0]) do
```
Comment (aadroher) [inline lib/modes/search.ex:27]: Yeah, maybe we need to agree on a common interface for the functions, i.e. if we want to define multiple signatures for the same name or go with default values (may not work in `PUSH`). Good for now, though!

File: test/modes/search_test.exs [hunk 2]
```diff
@@ -3,44 +3,84 @@ defmodule SonicClient.Modes.SearchTest do
   alias SonicClient.Modes.Search
   import SonicClient.TestConnectionHelper
 
+  setup_all do
+    add_data("user:1", "It is a common test")
+    add_data("user:2", "It is a common testable text")
+    add_data("user:3", "It should not appear in the common search")
+
+    on_exit(fn -> flush() end)
```
Comment (aadroher) [inline test/modes/search_test.exs:11]: Wait a minute, is `add_data` async, or something?

Comment (aadroher) [reply test/modes/search_test.exs:11]: Nah, I guess this is a test specific callback?

**Review by aadroher — APPROVED** [review]
This is great! I helped me a lot to understand how the thing works! 🙂

File: lib/modes/search.ex (hunk 1 above)
Comment (edgarlatorre) [reply lib/modes/search.ex:27]: Yeah we definitely need that. I don't have any strong opinion which one to use.

File: test/modes/search_test.exs (hunk 2 above)
Comment (edgarlatorre) [reply test/modes/search_test.exs:11]: `setup_all` is the test callback and `add_data` is function that is in the module `test/support/test_connection_helper.ex` which I'm importing in the test.

Comment (aadroher) [reply test/modes/search_test.exs:11]: @edgarlatorre I was referring to this `on_exit` function we pass an anonymous function to. I may be called at the end of the test suite?

Comment (edgarlatorre) [reply test/modes/search_test.exs:11]: Ah sorry, yes it's a callback which in the end of the the tests it calls the function which was passed as argument.

**Pull request merged**